org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterTuyaButton.java

```diff
@@ -0,0 +1,216 @@
+            } else {
+                logger.debug("{}: Error 0x{} setting server binding", endpoint.getIeeeAddress(),
+                        Integer.toHexString(bindResponse.getStatusCode()));
+                pollingPeriod = POLLING_PERIOD_HIGH;
+            }
+
+        } catch (InterruptedException | ExecutionException e) {
+            logger.error("{}: Exception setting reporting ", endpoint.getIeeeAddress(), e);
+        }
+
+        try {
+            CommandResult bindResponse = bind(clientCluster).get();
+            if (!bindResponse.isSuccess()) {
+                logger.error("{}: Error 0x{} setting client binding for cluster {}", endpoint.getIeeeAddress(),
+                        toHexString(bindResponse.getStatusCode()), ZclOnOffCluster.CLUSTER_ID);
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            logger.error("{}: Exception setting client binding to cluster {}: {}", endpoint.getIeeeAddress(),
+                        ZclOnOffCluster.CLUSTER_ID, e);
+        }
+
+        return true;
+    }
+
+    @Override
+    public synchronized boolean initializeConverter(ZigBeeThingHandler thing) {
+        super.initializeConverter(thing);
+
+        clientCluster = endpoint.getOutputCluster(ZclOnOffCluster.CLUSTER_ID);
+        serverCluster = endpoint.getInputCluster(ZclOnOffCluster.CLUSTER_ID);
+
+        if (clientCluster == null) {
+            logger.error("{}: Error opening device client controls", endpoint.getIeeeAddress());
+            return false;
+        }
+
+        if (serverCluster == null) {
+            logger.error("{}: Error opening device server controls", endpoint.getIeeeAddress());
+            return false;
+        }
+
+        clientCluster.addCommandListener(this);
+        serverCluster.addAttributeListener(this);
+
+        // Add Tuya-specific command
+        //
+        HashMap<Integer, Class<? extends ZclCommand>> commandMap = new HashMap<>();
+        commandMap.put(TuyaButtonPressCommand.COMMAND_ID, TuyaButtonPressCommand.class);
+        clientCluster.addClientCommands(commandMap);
+
+        return true;
+    }
+
+    @Override
+    public void disposeConverter() {
+        if(clientCluster != null) {
+            clientCluster.removeCommandListener(this);
+        }
+        if (serverCluster != null) {
+            serverCluster.removeAttributeListener(this);
+        }
+    }
+
+    @Override
+    public void handleRefresh() {
+        // nothing to do, as we only listen to commands
+    }
+
+    @Override
+    public Channel getChannel(ThingUID thingUID, ZigBeeEndpoint endpoint) {
+        // This converter is used only for channels specified in static thing types, and cannot be used to construct
+        // channels based on an endpoint alone.
+        return null;
+    }
+
+    @Override
+    public boolean commandReceived(ZclCommand command) {
+        logger.debug("{} received command {}", endpoint.getIeeeAddress(), command);
```

Comment:
Being petty, but please addd a `:` after the `{}` for the address - it's good to keep things standard as sometimes processing of logs looks for certain features.

Reply:
done, also, the others. I'll resolve without adding a separate comment to each.

Reply:
Thanks.